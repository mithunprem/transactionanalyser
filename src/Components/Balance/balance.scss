.balance {
  height: 50px;

  .balance-negative {
    color: #ff5e5e;
  }

  .balance-positive {
    color: #adff2f;
  }
}
