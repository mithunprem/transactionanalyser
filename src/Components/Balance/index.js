import React, { Fragment  } from 'react';

const Balance = ({ balance }) => {
  return (
    <Fragment>
      <p>{
          balance ?
            `Balance: $${String(balance.toFixed(2)).toLocaleString()}` :
            'Click calculate to show the balance.'
          }
      </p>
    </Fragment>
  )
}

export default Balance;
