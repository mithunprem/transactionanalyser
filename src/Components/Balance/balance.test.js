import React from 'react';
import { create } from 'react-test-renderer';
import Balance from '../../Components/Balance';

describe("Balance", () => {
  const successBalanceResult = {
    status : 'success',
    balance: 10,
    includedTransactions: 2
  }

  const errorBalanceResult = {
    status : 'error',
    message: 'No transactions found'
  }

  const defaultBalanceResult = {};

  test("should render without throwing an error for successful balance calculation", () => {
    const balance = create(<Balance balanceResult={successBalanceResult} />).toJSON()
    expect(balance).toMatchSnapshot()
  });

  test("should render without throwing an error for no transactions found scenario", () => {
    const balance = create(<Balance balanceResult={errorBalanceResult} />).toJSON()
    expect(balance).toMatchSnapshot()
  });

  test("should render without throwing an error for default case", () => {
    const balance = create(<Balance balanceResult={defaultBalanceResult} />).toJSON()
    expect(balance).toMatchSnapshot()
  });
});
