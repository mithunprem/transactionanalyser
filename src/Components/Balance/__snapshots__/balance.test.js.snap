// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance should render without throwing an error for default case 1`] = `
<p
  className="m-3 balance"
>
  Click calculate to show the balance
</p>
`;

exports[`Balance should render without throwing an error for no transactions found scenario 1`] = `
<p
  className="m-3 balance"
>
  No transactions found
</p>
`;

exports[`Balance should render without throwing an error for successful balance calculation 1`] = `
<p
  className="m-3 balance"
>
  <span>
    Relative balance for the period is: $10.00
  </span>
  <br />
  <span>
    Number of transactions included is: 2
  </span>
</p>
`;
